### PR TITLE
Undo 'any' inference propagation

### LINKED ISSUE #22362: Breaking change interpreting jquery promise types
**TypeScript Version:**  2.8.0-dev.20180302

**Code**

```ts
/// <reference types="jquery" /> // version 3.3.0
const t: JQuery.Promise2<string, Error, number, boolean, any, any> = {} as any;
$.when(t).then((a, b) => {
    const s: string = b; //should fail!
});
```

**Expected behavior:**

Error: `b` is `boolean`.

**Actual behavior:**

No error: `b` is `any`.

Has expected behavior in 2.8.0-dev.20180301, but behavior changed in 2.8.0-dev.20180302.
There are some additional error locations on DefinitelyTyped -- just run `dtslint --installAll` (to ensure you're using `typescript@next`), then `dtslint types jquery`. (The errors don't reproduce with just `tsc` because they come from `// $ExpectType` assertions.)

### PR DESCRIPTION
**Pull request by sandersn**
Fixes #22362 

#22197 introduced a change to inference where `any` as the source caused an inference from `any` to be added for all type parameters in the target. It's not clear that this "deep-any" behaviour is correct or useful.

Specifically, this change breaks inference in JQuery while changing only one other test; an inferred property deep inside mappedTypeRecursiveInference. There, when inferring from XMLHttpRequest to PartialDeep<T>, one property has type `any` which would otherwise have type `{}`.

In this PR, I just undid the change and added a test that @andy-ms simplified from the real jquery.d.ts on DefinitelyTyped.

### CODE REVIEW

**Comment by sandersn** [comment]
Note that inference pre-#22197 was not *right*; for `p1: MyPromise<number, U>`, we would always infer `p2: MyPromise<number, {}>` for all `U`. But at least we didn't infer `p2: MyPromise<any, any>` when starting with `p1: MyPromise<number, any>`.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11587,23 +11587,12 @@ namespace ts {
...
             let visited: Map<boolean>;
             let contravariant = false;
-            let propagationType: Type;
             inferFromTypes(originalSource, originalTarget);
 
             function inferFromTypes(source: Type, target: Type) {
                 if (!couldContainTypeVariables(target)) {
                     return;
                 }
-                if (source.flags & TypeFlags.Any) {
```
Comment (ahejlsberg) [inline src/compiler/checker.ts]: Change this to `source === wildcardType` and keep everything else unchanged.

**Comment by sandersn** [comment]
Sounds good. From discussion with @weswigham, it sounds like we need to revisit this after 2.8 to handle variance better.

**Comment by sandersn** [comment]
This is now in 2.8 as well.

**Pull request merged**